XenModel/Actions/ActionBase.cs

```diff
@@ -250,6 +249,11 @@ public ActionBase(string title, string description, bool suppressHistory, bool c
                 NewAction?.Invoke(this);
         }
 
+        public static ActionBase CreateDummyAction(string title, string description, bool completeImmediately = true, string error = null)
+        {
+            return new ActionBase(title, description, false, completeImmediately, error);
+        }
+

```

Comment:
This solution is definitely an improvement in terms of developer experience, but SonarQube is likely still going to complain. We're not using the return value of `CreateDummyAction` now.

Ideally we'd have a separate method invoke the actions we create, and we'd need to call it everywhere. Constructors shouldn't be performing these kinds of side effects.

something like `new ActionBase().Run()`